backend/memory/memory.go

```diff
@@ -368,14 +368,15 @@ func hasMatchingTag(tags map[string]string, itemTags map[string]string) bool {
 	return out
 }
 
-func (r *Backend) maintenance() {
+func (r *Backend) maintenance(alertList alerts.Alerts) {
```

Comment:
alerts should not know that we have different implementations for back-ends,
metrics should not know that we have different implementations for alerts [ or that we have any :-) ],

a. this `maintenance`  is for metrics, you should add a new  `maintenance` method in the `alerts` object.
b. the `open` is for metrics, you should add a new  `open` method in the `alerts` object.

Reply:
Just to make sure - in order for `checkAlerts` method to work we need to pass a `Backend` object to alerts so that it could query for metrics - is that okay with you? i.e to have the new method be `open(Backend, AlertsList)`?

Reply:
> method to work we need to pass a Backend object to alerts so that it could query for metrics

👍 

> is that okay with you?

That is exactly what I am thinking, see the api-server  -
https://github.com/MohawkTSDB/mohawk/blob/master/cmd/root.go#L132

> to have the new method be open(Backend, AlertsList)?

No, you create the object with the `db` object ( see L132 above ) and open with an `options` object that will include the alert-list, scan-interval and whatever ...